classes/task/check_recompletion.php

```diff
@@ -59,18 +59,26 @@ public function execute() {
             return;
         }
 
-        $sql = "SELECT cc.userid, cc.course
+        $sql = "SELECT cc.userid, cc.course,cc.timecompleted,r2.value as interval_spec
             FROM {course_completions} cc
             JOIN {local_recompletion_config} r ON r.course = cc.course AND r.name = 'enable' AND r.value = '1'
             JOIN {local_recompletion_config} r2 ON r2.course = cc.course AND r2.name = 'recompletionduration'
             JOIN {course} c ON c.id = cc.course
-            WHERE c.enablecompletion = ".COMPLETION_ENABLED." AND cc.timecompleted > 0 AND
-            (cc.timecompleted + ".$DB->sql_cast_char2int('r2.value').") < ?";
-        $users = $DB->get_recordset_sql($sql, array(time()));
+            WHERE c.enablecompletion = ".COMPLETION_ENABLED;
```

Comment:
we should still be checking if cc.timecompleted > 0 here - we don't want to be checking users that do not have a completion date.